ci/docker/linux-apt-r.dockerfile

```diff
@@ -81,25 +82,23 @@ RUN cat /arrow/ci/etc/rprofile >> $(R RHOME)/etc/Rprofile.site
 # Also ensure parallel compilation of C/C++ code
 RUN echo "MAKEFLAGS=-j$(R -s -e 'cat(parallel::detectCores())')" >> $(R RHOME)/etc/Renviron.site
 
-# Set up Python 3 and its dependencies
-RUN ln -s /usr/bin/python3 /usr/local/bin/python && \
-    ln -s /usr/bin/pip3 /usr/local/bin/pip
-
 COPY ci/scripts/r_deps.sh /arrow/ci/scripts/
 COPY r/DESCRIPTION /arrow/r/
 RUN /arrow/ci/scripts/r_deps.sh /arrow
 
-RUN pip install -U pip setuptools wheel
+ENV ARROW_PYTHON_VENV /arrow-dev
+COPY python/requirements-build.txt /arrow/python/
+RUN python3 -m venv ${ARROW_PYTHON_VENV} && \
+    source ${ARROW_PYTHON_VENV}/bin/activate && \
+    pip install -U pip setuptools wheel && \
+    pip install -r arrow/python/requirements-build.txt
```

Comment:
Am I reading this correctly that this ends up being effectively the same thing as 100-102 below, just into a venv?

Reply:
yes, the system Python on Ubuntu 24.04 forces you to pip install into a virtual env. That's why it fails without this.

Reply:
Ah, got it. Ok cool. If the tests pass (which they look like they do, this is good to go!)